crate/crash/tabulate.py

```diff
@@ -612,546 +29,46 @@ def _format(val, valtype, floatfmt, missingval="", has_invisible=True):
         tabulate(tbl, headers=hrow) == good_result
     True
 
-    """
+    """  # noqa
     if val is None:
         return missingval
 
-    if valtype in [int, _long_type, _text_type]:
+    if valtype in [int, _text_type]:
         return "{0}".format(val)
     elif valtype is _binary_type:
         try:
             return _text_type(val, "ascii")
         except TypeError:
             return _text_type(val)
     elif valtype is float:
-        is_a_colored_number = has_invisible and isinstance(val, (_text_type, _binary_type))
+        is_a_colored_number = has_invisible and isinstance(
+            val, (_text_type, _binary_type)
+        )
         if is_a_colored_number:
             raw_val = _strip_invisible(val)
             formatted_val = format(float(raw_val), floatfmt)
             return val.replace(raw_val, formatted_val)
+        # PATCH: Preserve string formatting even for numeric looking values.
+        # https://github.com/crate/crash/commit/1052e0d79
         elif not floatfmt:
-            return float_format(val)
+            return str(val)
```

Comment:
Apparently, this is the only patch we need to make the upstream version compatible with the previously vendored one. It looks pretty innocent, maybe there is a chance to bring it into mainline?